src/NuGet.Clients/NuGet.PackageManagement.UI/Options/AddMappingDialog.xaml

```diff
@@ -0,0 +1,89 @@
+<Window x:Class="NuGet.Options.AddMappingDialog"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:vsui="clr-namespace:Microsoft.VisualStudio.PlatformUI;assembly=Microsoft.VisualStudio.Shell.15.0"
+             xmlns:nuget="clr-namespace:NuGet.PackageManagement.UI"
+             mc:Ignorable="d" 
+             x:Uid="AddMappingDialogWindow"
+             AutomationProperties.Name="AddMappingDialogWindow"
+             Height="304"
+             Width="456"
+             ShowInTaskbar="False"
+             WindowStyle="None"
+             AllowsTransparency="True"
+             WindowStartupLocation="CenterOwner"
+             BorderBrush="{DynamicResource {x:Static vsui:EnvironmentColors.PanelBorderBrushKey}}"
+             BorderThickness="1">
+  <Grid>
+    <Grid.RowDefinitions>
+      <RowDefinition Height="30"/>
+      <RowDefinition Height="*"/>
+      <RowDefinition Height="36"/>
+    </Grid.RowDefinitions>
+    <StackPanel Grid.Row="0">
+      <Grid>
+        <Grid.ColumnDefinitions>
+          <ColumnDefinition Width="*"/>
+          <ColumnDefinition Width="50"/>
+        </Grid.ColumnDefinitions>
+        <TextBlock AutomationProperties.Name="AddNewPackageNamespace"
+                   Grid.Column="0"
+                   Text="{x:Static nuget:Resources.VSOptions_Label_AddPackageNamespace}"
+                   Margin="12,12,0,0"/>
+      </Grid>
+    </StackPanel>
+    <StackPanel Grid.Row="1">
+      <Grid>
+        <Grid.RowDefinitions>
+          <RowDefinition Height="48"/>
+          <RowDefinition Height="*"/>
+          <RowDefinition Height="36"/>
+        </Grid.RowDefinitions>
+        <vsui:WatermarkedTextBox Grid.Row="0" AutomationProperties.Name="Package ID" Name="packageID"  Margin="12,6,12,6" Watermark="{x:Static nuget:Resources.VSOptions_Watermark_AddPackageNamespace}" Background="White" Foreground="Black"/>
```

Comment:
The data eventually has a limit enforced on it for max length. The UI should also enforce that. 
(FYI, I tested this, and I can enter presumably infinity characters and it will save to the nuget.config, but later Restore fails validation 🧨) 
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMapping.cs#L16

Reply:
I added the MaxLength=100